zgw_consumers/contrib/setup_configuration/steps.py

```diff
@@ -0,0 +1,46 @@
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import SelfTestFailed
+from requests.exceptions import RequestException
+
+from zgw_consumers.client import build_client
+from zgw_consumers.models import Service
+
+from .models import ServicesConfigurationModel
+
+
+class ServiceConfigurationStep(BaseConfigurationStep[ServicesConfigurationModel]):
+    """
+    Configure admin login via OpenID Connect
+    """
+
+    verbose_name = "Configuration to connect with external services"
+    config_model = ServicesConfigurationModel
+    namespace = "ZGW_CONSUMERS"
+    enable_setting = "ZGW_CONSUMERS_CONFIG_ENABLE"
+
+    def is_configured(self, model) -> bool:
+        slugs = [config.slug for config in model.services]
+        return Service.objects.filter(slug__in=slugs).count() == len(slugs)
+
+    def execute(self, model):
+        for config in model.services:
+            Service.objects.update_or_create(
+                slug=config.slug,
+                defaults={k: v for k, v in config.dict().items() if k != "slug"},
+            )
+
+    def validate_result(self, model) -> None:
+        slugs = [config.slug for config in model.services]
+        exceptions = []
+        for service in Service.objects.filter(slug__in=slugs):
+            client = build_client(service)
+            try:
+                response = client.get("")
```

Comment:
shouldn't this use the model-level connection test? Additionally, in the YAML you could include an expected status code to validate with the connection test, since at that point you tend to have semantic information that the library in generic form doesn't have

Reply:
You're right! I've replace it and added an `connection_check_expected_status` field to configure this